lightwood/encoder/numeric/numeric.py

```diff
@@ -1,46 +1,63 @@
 from lightwood.helpers.log import log
 from lightwood.helpers.general import is_none
+from lightwood.api.dtype import dtype
 
 
 class NumericEncoder(BaseEncoder):
+    """
+    The numeric encoder takes numbers (float or integer) and converts it into tensors of the form:
+    ``[is_none(number), is_positive(number), natural_log(abs(number)), number/absolute_mean]``
 
-    def __init__(self, data_type=None, is_target: bool = False, positive_domain: bool = False):
+    This representation is: ``[is_positive(number), natural_log(abs(number)), number/absolute_mean]]`` if encoding target values, since target values can't be none.
+
+    The ``absolute_mean`` is computed in the ``prepare`` method and is just the mean of the absolute values of all numbers feed to prepare (which are not none)
+
+    ``none`` stands for any number that is an actual python ``None`` value or any sort of non-numeric value (a string, nan, inf)
+    """ # noqa
+
+    def __init__(self, data_type: dtype = None, is_target: bool = False, positive_domain: bool = False):
+        """
+        :param data_type: The data type of the number (integer, float, quantity)
+        :param is_target: If we are encoding a target value or an input value (feature)
+        :param positive_domain: Forces the encoder to always output positive values
+        """
         super().__init__(is_target)
         self._type = data_type
         self._abs_mean = None
         self.positive_domain = positive_domain
         self.decode_log = False
         self.output_size = 4 if not self.is_target else 3
 
-    def prepare(self, priming_data):
+    def prepare(self, priming_data: Iterable):
+        """
+        :param priming_data: an iterable data structure containing numbers numbers which will be used to compute the values used for normalizing the encoded representations
+        """ # noqa
         if self.is_prepared:
             raise Exception('You can only call "prepare" once for a given encoder.')
 
         value_type = 'int'
         for number in priming_data:
-            try:
-                number = float(number)
-            except Exception:
-                continue
-
-            if np.isnan(number):
-                err = 'Lightwood does not support working with NaN values !'
-                log.warning(err)
-                continue
-
-            if int(number) != number:
-                value_type = 'float'
+            if not is_none(number):
+                if int(number) != number:
+                    value_type = 'float'
 
         self._type = value_type if self._type is None else self._type
-        non_null_priming_data = [float(str(x).replace(',', '.')) for x in priming_data if not is_none(x)]
+        non_null_priming_data = [x for x in priming_data if not is_none(x)]
         self._abs_mean = np.mean(np.abs(non_null_priming_data))
         self.is_prepared = True
 
-    def encode(self, data):
+    def encode(self, data: Iterable):
```

Comment:
describe the process

Reply:
I feel like this is already well described under `NumeriEncoder` (indeed, too well, the fallacy of writing down code as comments, the code will change and the comments won't, the code/comments will have bugs that the other won't)

I can move that description here, or otherwise, feel free to suggest what's missing from the description.